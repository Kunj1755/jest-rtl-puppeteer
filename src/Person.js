import React, { useState } from "react";
const person = props => {
    const [count, setCount] = useState(0);
    const _onClick = () => {
        setCount(count + 1);
    }
    return (
        <div>
            <p onClick={_onClick}>
                I'm a person and My name is {props.name} and I am {props.age} years old
      </p>
            <p> {props.children}</p>
            <p >{count}</p>
            <p id="para">100</p>
            <input type="text" onChange={props.changed} />
        </div>
    );
};
export default person;
