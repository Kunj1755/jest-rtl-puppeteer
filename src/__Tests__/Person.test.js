import React from "react";
import TestRenderer from "react-test-renderer";
import { cleanup, fireEvent, render } from '@testing-library/react';
import Person from "../Person";

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

const personCompProps = {
    name: "Ram",
    age: "30",
    handleOnClick: jest.fn(),
    handleOnChange: jest.fn()

};

test("Person renders without crashing", () => {
    const testRenderer = TestRenderer.create(<Person>I am a person</Person>);
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Person renders without crashing 2", () => {
    const testRenderer = TestRenderer.create(<Person>I am a person</Person>);
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Person renders with name and age props", () => {
    const testRenderer = TestRenderer.create(<Person name={personCompProps.name}
        age={personCompProps.age}>I am a person</Person>);
    const tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});

test.only("onClick handler called", () => {
    // const handleOnClick = jest.fn();
    const testRenderer = TestRenderer.create(<Person name={personCompProps.name}
        age={personCompProps.age} click={personCompProps.handleOnClick}>I am a person</Person>);
    // const tree = testRenderer.toJSON();
    // console.log(testRenderer.toJSON());
    // tree.props.onClick();
    // expect(tree).toMatchSnapshot();
});
