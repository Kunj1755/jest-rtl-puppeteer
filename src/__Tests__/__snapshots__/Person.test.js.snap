// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Person renders with name and age props 1`] = `
<div>
  <p
    onClick={[Function]}
  >
    I'm a person and My name is 
    Ram
     and I am 
    30
     years old
  </p>
  <p>
     
    I am a person
  </p>
  <p>
    0
  </p>
  <input
    type="text"
  />
</div>
`;

exports[`Person renders without crashing 1`] = `
<div>
  <p
    onClick={[Function]}
  >
    I'm a person and My name is 
     and I am 
     years old
  </p>
  <p>
     
    I am a person
  </p>
  <p>
    0
  </p>
  <input
    type="text"
  />
</div>
`;

exports[`Person renders without crashing 2 1`] = `
<div>
  <p
    onClick={[Function]}
  >
    I'm a person and My name is 
     and I am 
     years old
  </p>
  <p>
     
    I am a person
  </p>
  <p>
    0
  </p>
  <input
    type="text"
  />
</div>
`;
